extensions/wikia/Search/classes/Services/CombinedSearchService.php

```diff
@@ -213,182 +266,215 @@ public function queryPhraseForWikias( $query, $hubs, $lang ) {
-			')) AND +((title_en:"' . $query . '") OR (redirect_titles_mv_en:"' . $query . '")) AND +(nolang_txt:"' . $query . '")';
+		$query = '+(' .
+			$hubQuery .
+			'(' .
+			implode( ' OR ', $nsArr ) .
+			') AND (lang:' .
+			$lang .
+			')) AND +((title_en:"' .
+			$query .
+			'") OR (redirect_titles_mv_en:"' .
+			$query .
+			'")) AND +(nolang_txt:"' .
+			$query .
+			'")';
+
 		return $query;
 	}
 
 	/**
 	 * @param $query
+	 *
 	 * @return mixed|string
 	 */
 	protected function sanitizeQuery( $query ) {
 		$select = new Select( $query );
+
 		return $select->getSolrQuery( 10 );
 	}
 
 	/**
 	 * @param $query
 	 * @param $langs
 	 * @param $hubs
+	 *
 	 * @return array
 	 */
 	public function searchForWikias( $query, $langs, $hubs ) {
 		$timer = Time::start( [ __CLASS__, __METHOD__ ] );
-		$wikias = [ ];
+		$wikias = [];
 		foreach ( $langs as $lang ) {
 			$crossWikiResults = $this->queryForWikias( $query, $hubs, $lang );
 			foreach ( $crossWikiResults as $wikiSearchResult ) {
-				$wikias[ ] = $this->processWiki( $wikiSearchResult );
+				$wikias[] = $this->processWiki( $wikiSearchResult );
 			}
 			if ( sizeof( $wikias ) >= self::CROSS_WIKI_RESULTS ) {
 				break;
 			}
 		}
 		$wikias = array_slice( $wikias, 0, self::CROSS_WIKI_RESULTS );
 		$timer->stop();
+
 		return $wikias;
 	}
 
 	public function phraseSearchForWikias( $query, $langs, $hubs ) {
 		$timer = Time::start( [ __CLASS__, __METHOD__ ] );
-		$wikias = [ ];
+		$wikias = [];
 		foreach ( $langs as $lang ) {
 			$crossWikiResults = $this->queryPhraseForWikias( $query, $hubs, $lang );
 			foreach ( $crossWikiResults as $wikiSearchResult ) {
 				//PLA-1116: set to false
-				$wikias[ ] = $this->processWiki( $wikiSearchResult, false );
+				$wikias[] = $this->processWiki( $wikiSearchResult, false );
 			}
 			if ( sizeof( $wikias ) >= self::CROSS_WIKI_RESULTS ) {
 				break;
 			}
 		}
 		$timer->stop();
+
 		return $wikias;
 	}
 
 	/**
 	 * @param $query
 	 * @param $hubs
 	 * @param $lang
+	 *
 	 * @return array
 	 */
 	public function queryForWikias( $query, $hubs, $lang ) {
 		$crossWikiSearchConfig = new Config;
-		$crossWikiSearchConfig->setQuery( $query )
-			->setLimit( self::CROSS_WIKI_RESULTS )
-			->setPage( 1 )
-			->setRank( 'default' )
-			->setInterWiki( true )
-			->setCommercialUse( $this->getHideNonCommercialContent() )
-			->setLanguageCode( $lang );
+		$crossWikiSearchConfig->setQuery( $query )->setLimit( self::CROSS_WIKI_RESULTS )->setPage( 1 )->setRank(
+				'default'
+			)->setInterWiki( true )->setCommercialUse( $this->getHideNonCommercialContent() )->setLanguageCode( $lang );
 		if ( !empty( $hubs ) ) {
 			$crossWikiSearchConfig->setHubs( $hubs );
 		}
 		$crossWikiResultSet = ( new Factory )->getFromConfig( $crossWikiSearchConfig )->search();
-		$crossWikiResults = $crossWikiResultSet->toArray( [ "sitename_txt", "url", "id", "description_txt", "lang_s", "score", "description_txt" ] );
+		$crossWikiResults = $crossWikiResultSet->toArray(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$crossWikiResults". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)